orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java

```diff
@@ -0,0 +1,228 @@
+package com.yahoo.ycsb.db;
+
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+import com.orientechnologies.orient.core.dictionary.ODictionary;
+import com.orientechnologies.orient.core.record.ORecordInternal;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.StringByteIterator;
+
+import org.junit.*;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by kruthar on 12/29/15.
+ */
+public class OrientDBClientTest {
+  // TODO: This must be copied because it is private in OrientDBClient, but this should defer to table property.
+  private static final String CLASS = "usertable";
+  private static final int FIELD_LENGTH = 32;
+  private static final String FIELD_PREFIX = "FIELD";
+  private static final String KEY_PREFIX = "user";
+  private static final int NUM_FIELDS = 3;
+  private static final String TEST_DB_URL = "memory:test";
+
+  private static ODatabaseDocumentTx orientDBConnection = null;
+  private static ODictionary<ORecordInternal<?>> orientDBDictionary;
+  private static OrientDBClient orientDBClient = null;
+
+  @Before
+  public void setup() {
+    try {
+      orientDBClient = new OrientDBClient();
+
+      Properties p = new Properties();
+      // TODO: Extract the property names into final variables in OrientDBClient
+      p.setProperty("orientdb.url", TEST_DB_URL);
+
+      orientDBClient.setProperties(p);
+      orientDBClient.init();
+      orientDBConnection = new ODatabaseDocumentTx(TEST_DB_URL).open("admin","admin");
+      orientDBDictionary = orientDBConnection.getMetadata().getIndexManager().getDictionary();
+    } catch (DBException e) {
+      e.printStackTrace();
+      fail("Could not create OrientDBClient instance");
+    }
+  }
+
+  @After
+  public void teardown() {
+    if (orientDBConnection != null) {
+      orientDBConnection.close();
+    }
+
+    try {
+      if (orientDBClient != null) {
+        orientDBClient.cleanup();
+      }
+    } catch (DBException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /*
+      This is a copy of buildDeterministicValue() from core:com.yahoo.ycsb.workloads.CoreWorkload.java.
+      That method is neither public nor static so we need a copy.
+   */
+  private String buildDeterministicValue(String key, String fieldkey) {
```

Comment:
Thinking about this more, I agree that copying makes the most sense right now. It probably points to a need to come up with a common strategy for these kinds of tests, but that's not an issue for this PR.